assets/backport/github/github.go

```diff
@@ -0,0 +1,398 @@
+	err = c.updateBranch(ctx, branchName, sha)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	return mergeTree, sha, nil
+}
+
+// createSiblingCommit creates a commit with the passed in commit's tree and parent
+// and updates the passed in branch to point at that commit.
+func (c *Client) createSiblingCommit(ctx context.Context, branchName string, branchHeadCommit *go_github.Commit, cherryParent *go_github.Commit) error {
+	tree := branchHeadCommit.GetTree()
+	// This will be the "temp" commit, commit is lost. Commit message doesn't matter.
+	commitSHA, err := c.createCommit(ctx, "temp", tree, cherryParent)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	err = c.updateBranch(ctx, branchName, commitSHA)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// createBranchFrom creates a branch from the passed in branch's HEAD.
+func (c *Client) createBranchFrom(ctx context.Context, branchFromName string, newBranchName string) error {
+	baseBranch, err := c.getBranch(ctx, branchFromName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	newRefBranchName := fmt.Sprintf("%s%s", branchRefPrefix, newBranchName)
+	baseBranchSHA := baseBranch.GetCommit().GetSHA()
+
+	ref := &go_github.Reference{
+		Ref: &newRefBranchName,
+		Object: &go_github.GitObject{
+			SHA: &baseBranchSHA, /* SHA to branch from */
+		},
+	}
+	_, _, err = c.Client.Git.CreateRef(ctx, c.Organization, c.Repository, ref)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// updateBranch updates a branch.
+func (c *Client) updateBranch(ctx context.Context, branchName string, sha string) error {
+	refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+	_, _, err := c.Client.Git.UpdateRef(ctx, c.Organization, c.Repository, &go_github.Reference{
+		Ref: &refName,
+		Object: &go_github.GitObject{
+			SHA: &sha,
+		},
+	}, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// getBranch gets a branch.
+func (c *Client) getBranch(ctx context.Context, branchName string) (*go_github.Branch, error) {
+	branch, _, err := c.Client.Repositories.GetBranch(ctx, c.Organization, c.Repository, branchName, true)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return branch, nil
+}
+
+// createCommit creates a new commit.
+func (c *Client) createCommit(ctx context.Context, commitMessage string, tree *go_github.Tree, parent *go_github.Commit) (string, error) {
+	commit, _, err := c.Client.Git.CreateCommit(ctx, c.Organization, c.Repository, &go_github.Commit{
+		Message: &commitMessage,
+		Tree:    tree,
+		Parents: []*go_github.Commit{
+			parent,
+		},
+	})
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return commit.GetSHA(), nil
+}
+
+// getCommit gets a commit.
+func (c *Client) getCommit(ctx context.Context, sha string) (*go_github.Commit, error) {
+	commit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		sha)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return commit, nil
+}
```

Comment:
Wrapping these function calls within another function without doing anything else does not have much of a benefit. Please remove them here and elsewhere where you are doing this.

I did something similar in the `github` package within the bot, but it was for two reasons:

1. I was using it to fake the response from the GitHub API in tests. For example this would allow me to control what `ListFiles` returns in tests. I don't see you doing this.
2. It would make it easier to use paginated APIs. Using wrappers for pagination is fine.

Since in this situation I don't see you using this in tests, it's better to just remove the wrappers.